### DISCUSSION TITLE
How to improve best bound in CP-SAT?

### DISCUSSION THREAD

**Discussion by C8504** [google/or-tools#3363] [CP-SAT questions] [answered]
Hi Experts,

I am using CP-SAT solver, I know the optimal solution is 3813.
According to below log info, 
the best solution is almost optimal, while the bound is much far from optimal solution.
Could I know any parameters/algorithms to improve the bound in CP-SAT? 

Thanks, John.

#47    412.75s best:4027.22222 next:[2137.77778,4026.66667] rnd_var_lns_default(d=0.75 s=107 t=0.10 p=0.53)
#Bound 425.13s best:4027.22222 next:[2168.33333,4026.66667] max_lp
#Bound 432.87s best:4027.22222 next:[2201.11111,4026.66667] max_lp
#Bound 448.01s best:4027.22222 next:[2204.44444,4026.66667] max_lp
#48    461.49s best:4021.11111 next:[2204.44444,4020.55556] rnd_var_lns_default(d=0.75 s=117 t=0.10 p=0.52)
#Bound 462.20s best:4021.11111 next:[2243.33333,4020.55556] max_lp
#Bound 476.52s best:4021.11111 next:[2273.33333,4020.55556] max_lp

**Comment by lperron** [comment, marked as answer]
This is unfortunately one of the most open questions I have seen.
I work on this all the time :-)

Try to add redundant constraints. But it is well know that many classes of problems have very bad lower bounds.

**Reply by C8504** [reply]
thanks lperron.
Yesterday, I used mip solver to solve this problem as well, the mip solver gave me the lp relaxation solution equals to 3500 at root node.
I don't know if we could use glop more heavy to get the lp solution as better bound to guide.

John

**Reply by lperron** [reply]
can you send me the proto ?

**Reply by lperron** [reply]
if you are not doing so, using 16 workers should help a little

**Reply by C8504** [reply]
ok, I will try.Thanks again.
